sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala

```diff
@@ -114,6 +117,42 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
       (createCode, extract + patch)
     }.unzip
 
+    /*
+     * 200 = 6000 bytes / 30 (up to 30 bytes per one call))
+     * the maximum byte code size to be compiled for HotSpot is 8000.
+     * We should keep less than 8000
+     */
+    val numberOfStatementsThreshold = 200
+    val (initializerAccessorFuncs, initializerAccessorCalls, extractorFuncs, extractorCalls) =
+      if (initializeAccessors.length <= numberOfStatementsThreshold) {
```

Comment:
Yes we could do. But, (another PR)[https://github.com/apache/spark/pull/7076#issuecomment-122176653] intentionally avoid putting them into a group if they can be put into one group.
Which is better?